### PR TITLE
Refactor redshift & postgres crawlers using system catalogs [sc-5078]

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

Using system catalogs to retrieve dataset information.

### 🤓 What?

Rewrite `_fetch_table`, `_fetch_columns` and `_fetch_constraints`.

### 🧪 Tested?

Manually test postgres and redshift connector, no different after refactoring.

### CODE REVIEW

**Review by mars-lan — COMMENTED** [review]
Please also update README with the new persimmon requirements

File: metaphor/postgresql/README.md [hunk 1]
```diff
@@ -4,9 +4,13 @@ This connector extracts technical metadata from a PostgreSQL database using [asy
...
 
-Currently you must run the connector using a user with `SELECT` [privilege](https://www.postgresql.org/docs/current/ddl-priv.html) to all tables & views.
+You must run the connector using a user with ``SELECT` [privilege](https://www.postgresql.org/docs/current/ddl-priv.html) to the following tables:
 
-> We're working on extracting the metadata from [system catalogs](https://www.postgresql.org/docs/current/catalogs.html), which greatly reduces the privileges required for the user.
+- `pg_catalog.pg_constraint`
+- `pg_catalog.pg_class`
+- `pg_catalog.pg_namespace`
+- `pg_catalog.pg_attribute`
+- `pg_catalog.pg_description`
```
Comment (mars-lan) [inline metaphor/postgresql/README.md:13]: Please also include the corresponding `GRANT` statement to make it easy for someone without strong Postgres knowledge.

**Comment by mars-lan** [comment]
Please also update https://github.com/MetaphorData/connectors/tree/main/metaphor/redshift

File: metaphor/postgresql/README.md (hunk 1 above)
Comment (elic-eon) [reply metaphor/postgresql/README.md:13]: Updated.

File: metaphor/postgresql/README.md [hunk 2]
```diff
@@ -4,9 +4,19 @@ This connector extracts technical metadata from a PostgreSQL database using [asy
...
+You must run the connector using a user with `SELECT` [privilege](https://www.postgresql.org/docs/current/ddl-priv.html) to the following tables:
 
-> We're working on extracting the metadata from [system catalogs](https://www.postgresql.org/docs/current/catalogs.html), which greatly reduces the privileges required for the user.
+- `pg_catalog.pg_constraint`
+- `pg_catalog.pg_class`
+- `pg_catalog.pg_namespace`
+- `pg_catalog.pg_attribute`
+- `pg_catalog.pg_description`
+
+Or, use the command below:
```
Comment (mars-lan) [inline metaphor/postgresql/README.md:15]:
```suggestion
Or, use the following command to grant the privileges:
```

File: metaphor/postgresql/README.md [hunk 3]
```diff
@@ -4,9 +4,19 @@ This connector extracts technical metadata from a PostgreSQL database using [asy
...
+
+```sql
+GRANT SELECT ON pg_catalog.pg_constraint, pg_catalog.pg_class, pg_catalog.pg_namespace, pg_catalog.pg_attribute, pg_catalog.pg_description TO [User]
```
Comment (mars-lan) [inline metaphor/postgresql/README.md:18]:
```suggestion
GRANT SELECT ON pg_catalog.pg_constraint, pg_catalog.pg_class, pg_catalog.pg_namespace, pg_catalog.pg_attribute, pg_catalog.pg_description TO <user>
```

File: metaphor/redshift/README.md [hunk 4]
```diff
@@ -4,9 +4,7 @@ This connector extracts technical metadata from a Redshift database using [async
 
 ## Setup
 
-Currently you must run the connector using a user with `SELECT` [privilege](https://docs.aws.amazon.com/redshift/latest/dg/r_Privileges.html) to all tables & views.
-
-> We're working on extracting the metadata from [system catalogs](https://docs.aws.amazon.com/redshift/latest/dg/c_intro_catalog_views.html), which greatly reduces the privileges required for the user.
+The connector extracts the metadata from [system catalogs](https://docs.aws.amazon.com/redshift/latest/dg/c_intro_catalog_views.html), with restricted access to system tables and additional `SELECT` [privilege](https://www.postgresql.org/docs/current/ddl-priv.html) to `pg_catalog.svv_table_info`. Use `GRANT SELECT ON pg_catalog.svv_table_info TO [User];` to grant permission.
```
Comment (mars-lan) [inline metaphor/redshift/README.md:7]:
```suggestion
The connector extracts the metadata from [system catalogs](https://docs.aws.amazon.com/redshift/latest/dg/c_intro_catalog_views.html), with restricted access to system tables and additional `SELECT` [privilege](https://www.postgresql.org/docs/current/ddl-priv.html) to `pg_catalog.svv_table_info`.

Use the following command to grant the permission:

``` 
GRANT SELECT ON pg_catalog.svv_table_info TO <user>;
```
```

Comment (mars-lan) [reply metaphor/redshift/README.md:7]: Embedding code block within suggestion doesn't work well but hopefully you get my point.

**Review by mars-lan — COMMENTED** [review]
Don't forget to bump up the version.

**Review by mars-lan — APPROVED** [review]
Great work!

**Pull request merged**